testsuite/tests/typing-local/nosyntax.ml

```diff
@@ -30,3 +30,33 @@ let local_ref (f : lfn -> unit) =
 [%%expect{|
 val local_ref : (lfn -> unit) -> unit = <fun>
 |}]
+
+
+type foo = Foo of string
+type gfoo = GFoo of (string [@ocaml.global])
+(* native global_ syntax is not supported yet *)
+(* won't parse *)
+
+[%%expect{|
+type foo = Foo of string
+type gfoo = GFoo of (string [@global])
+|}]
+
+let cast1 ((x : foo) [@ocaml.local]) = match x with
+  Foo s -> GFoo s
+[%%expect{|
+Line 2, characters 16-17:
+2 |   Foo s -> GFoo s
+                    ^
+Error: This value escapes its region
+|}]
+
+
+let cast2 ((x : gfoo) [@ocaml.local]) = match x with
+  GFoo s -> Foo s
+[%%expect{|
+val cast2 : (gfoo [@local]) -> foo = <fun>
+|}]
+
```

Comment:
I'm not sure why these tests are here. Instead I would have expected to see some tests which print `[@global]` in error messages e.g. errors for incorrect module inclusion around global fields.

Reply:
Currently both record fields and constructor arguments are broken in this regards when local extension disabled: they would accept `global_` syntax without error; and record fields in particular also prints `global_` instead of `[@global]`. I suggest that I make another patch fixing record fields, then base this patch on that.